prefect_github/filesystem.py

```diff
@@ -0,0 +1,117 @@
+"""The GitHub filesystem for Prefect"""
+import io
+import tempfile
+from pathlib import Path
+from typing import Optional
+
+from prefect.filesystems import ReadableDeploymentStorage
+from prefect.utilities.processutils import run_process
+from pydantic import Field, validator
+
+from prefect_github import GitHubCredentials
+from prefect_github.exceptions import InvalidRepositoryURLError
+
+
+class GitHubRepository(ReadableDeploymentStorage):
+    """
+    Interact with files stored on GitHub repositories.
+    """
+
+    _block_type_name = "GitHubRepository"
+    _logo_url = "https://images.ctfassets.net/gm98wzqotmnx/187oCWsD18m5yooahq1vU0/ace41e99ab6dc40c53e5584365a33821/github.png?h=250"  # noqa: E501
+
+    repository: str = Field(
+        default=...,
+        description=(
+            "The URL of a GitHub repository to read from, in either HTTPS or SSH "
+            "format. If you are using a private repo, it must be in the HTTPS format."
+        ),
+    )
+    reference: Optional[str] = Field(
+        default=None,
+        description="An optional reference to pin to; can be a branch name or tag.",
+    )
+    credentials: Optional[GitHubCredentials] = Field(
+        default=None,
+        description="An optional GitHubCredentials block for using private GitHub repos.",  # noqa: E501
+    )
+
+    @validator("credential")
+    def ensure_credentials_go_with_https(cls, v: str, values: dict):
+        """Ensure that credentials are not provided with 'SSH' formatted GitHub URLs."""
+        if v is not None:
+            if values["repository"].startswith("git@"):
```

Comment:
I think this might match the error of "using the 'HTTPS' format" more (and also catch cases of `http://`
```suggestion
            if not values["repository"].startswith("https"):
```

Reply:
Very reasonable